core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java

```diff
@@ -91,27 +68,44 @@ public void close() {
         try {
             ((CloseableHttpClient)getHttpClient()).close();
         } catch (IOException e) {
-            InternalLogger.INSTANCE.error("Failed to close http client, exception: %s", e.toString());
+            InternalLogger.INSTANCE.error("Failed to close http httpClient, exception: %s", e.toString());
+        } finally {
+            synchronized (lock) {
+                if (connectionManager != null) {
+                    connectionManager.shutdown();
+                }
+            }
         }
     }
 
     @Override
     public HttpClient getHttpClient() {
```

Comment:
why not just return CloseableHttpClient here and prevent the cast in Line 69?

Reply:
This is the interface implementation. The interface has to supports 4.2 and 4.3. `CloseableHttpClient` does not exist in 4.2

Reply:
I see makes sense.